grails-app/domain/org/broadinstitute/orsp/Issue.groovy

```diff
@@ -215,8 +217,47 @@ class Issue implements LogicalDelete<Issue> {
      * @return True if all attachments have the 'Approved' status, false otherwise
      */
     transient Boolean attachmentsApproved() {
-        getAttachments()?.collect {
-            it.status
-        }?.findAll{ it != IssueStatus.Approved.getName() }?.isEmpty()
+        Boolean completed = false
+        Boolean optionalKey = true
+
+        ArrayList approvedTypeDocuments = getAttachments().findAll {
+            it.status == IssueStatus.Approved.getName()
+        }.fileType
+
+
+        if (getType() == IssueType.CONSENT_GROUP.name) {
```

Comment:
This complex if-else-if-else ... etc would read a lot better if it were a switch on `IssueType`